### PR TITLE
Stable version fixing most major bugs in test cases

### PR DESCRIPTION
**Pull request by kdobolyi**

### CODE REVIEW

File: geolocation.py [hunk 1]
```diff
@@ -56,6 +56,9 @@ def edit_distance_to_world(location):
     # pylint: disable=invalid-name
 
     # special cities and countries
+
+    location = location.title()
```
Comment (jspeed-meyers) [inline geolocation.py:60]: @kdobolyi, can you please put an explanatory note here? I don't know what this title method does.

File: geolocation.py [hunk 2]
```diff
@@ -106,6 +109,51 @@ def get_country_from_location(location_string):
     if location_string is None:
         return "None"
 
+    location_string = location_string[0].upper() + location_string[1:]
```
Comment (jspeed-meyers) [inline geolocation.py:112]: @kdobolyi, does capitalize() fit your use case here? See: https://docs.python.org/3/library/stdtypes.html#str.capitalize - Feel free to ignore if it does not.

File: geolocation.py [hunk 3]
```diff
@@ -106,6 +109,51 @@ def get_country_from_location(location_string):
...
+            return "None"
+
+    special_locations_international = {'Scotland':'United Kingdom'}
+    for special in special_locations_international.keys():
+        if special in location_string:
+            return special_locations_international[special]
+
+    location_string.encode("utf-8")
+
+    # one of the weird cities that has larger North American populatoin than global population
```
Comment (jspeed-meyers) [inline geolocation.py:136]: Small typo in comment (populatoin --> population).

File: geolocation.py [hunk 4]
```diff
@@ -137,15 +187,22 @@ def get_country_from_location(location_string):
         # Check different positions of the token
         for position in [-1, 0]:
             pieces = location_string.split(separator)
+
             token = pieces[position].strip()
+            if len(token) > 2:
```
Comment (jspeed-meyers) [inline geolocation.py:192]: @kdobolyi, can you please put an explanatory comment here? It wasn't immediately obvious to me what is going on.

File: geolocation.py [hunk 5]
```diff
@@ -137,15 +187,22 @@ def get_country_from_location(location_string):
...
             elif token in ALL_COUNTRIES:  # pylint: disable=no-else-return
                 return token
             elif token in CODE_COUNTRY_DICT.keys():
                 return CODE_COUNTRY_DICT[token]
+            elif token in METRO_AREA_COUNTRY_DICT.keys():
+                return METRO_AREA_COUNTRY_DICT[token]
 
     return edit_distance_to_world(location_string)
+
+#print(get_country_from_location("Sri-City, Andhra Pradesh"))
```
Comment (jspeed-meyers) [inline geolocation.py:208]: Please remove this dead code.

**Review by jspeed-meyers — CHANGES_REQUESTED** [review]
Great work, @kdobolyi, on adding these tests and making the country prediction functionality more capable. Some minor requests.

File: geolocation.py (hunk 2 above)
Comment (kdobolyi) [reply geolocation.py:112]: almost -- that one will force the rest to be lowercase, which in some cases may not be what we want :-) (I could probably write cleaner code that just lowercases everything, but that's a rewrite for another day :-))

**Review by jspeed-meyers — APPROVED** [review]
Nice!

**Comment by jspeed-meyers** [comment]
Merge when you're ready! Consider squashing the commits.

**Pull request merged**